src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs

```diff
@@ -119,6 +127,9 @@ public QueueProcessor(QueueProcessorFactoryContext context)
                 if (message.DequeueCount >= MaxDequeueCount)
                 {
                     await CopyMessageToPoisonQueueAsync(message, _poisonQueue, cancellationToken);
+
+                    message.SetProperties(id, popReceipt, insertionTime, nextVisibleTime, expirationTime);
```

Comment:
I've built the conditional into SetProperties in my next update. It'll only update values if they've changed.